Assets/Scripts/Localization/LocalizationTable.cs

```diff
@@ -6,193 +6,118 @@
+        /// <summary>
+        /// Load a localization file from the harddrive with a defined localization code.
+        /// </summary>
+        /// <param name="path">The path to the file.</param>
+        /// <param name="localizationCode">The localization code, e.g.: "en_US", "en_UK"</param>
+        private static void LoadLocalizationFile(string path, string localizationCode)
         {
-            // Read the contents of the file. This might throw an exception!
-            string[] lines = File.ReadAllLines(path);
-
-            // Create an empty char array outside the foreach loop, for optimization reasons.
-            // This is used as a storage for the chars in the line.
-            char[] chars;
-
-            // The key that the loop assembled.
-            string currentKey = "";
-
-            // The value that the loop assembled.
-            string currentValue = "";
-
-            // Is the loop already done with figuring out the key?
-            bool searchingValue = false;
-
-            foreach (string line in lines)
+            try
             {
-                // Reuse the array (for reducing RAM usage).
-                chars = line.ToCharArray();
-
-                // Set searching value to false.
-                searchingValue = false;
-
-                // Set the key to an empty string.
-                currentKey = "";
-
-                // Set the value to an empty string.
-                currentValue = "";
-
-                // Go through each char contained in this line.
-                foreach (char c in chars)
+                localizationTable[localizationCode] = new Dictionary<string, string>();
+                string[] lines = File.ReadAllLines(path);
+                foreach (string line in lines)
                 {
-                    // Check if the loop is searching for a value.
-                    if (!searchingValue)
+                    string[] keyValuePair = line.Split(new char[] { '=' }, 2);
+                    if (keyValuePair.Length != 2)
                     {
-                        // Check if the current char is an '=', if not, add the char to the key.
-                        if (c != '=')
-                        {
-                            // Add the char to the key.
-                            currentKey += c;
-                        }
-                        else
-                        {
-                            // The char is an '=', set searchingValue to true and ignore the current char.
-                            searchingValue = true;
-                        }
-                    }
-                    else
-                    {
-                        // The loop is searching for a value. Add the current char, regardless of what it is.
-                        currentValue += c;
+                        throw new InvalidOperationException(string.Format("Invalid format of localization string. Actual {0}", line));
```

Comment:
Yes, yes, yes I couldnt agree more. Crash early, crash often, I say. Maybe this should be discussed and settle on a standard. I have a feeling we are the minority. :P